antarest/study/common/studystorage.py

```diff
@@ -47,6 +77,60 @@ def copy(self, src_meta: T, dest_meta: T, with_outputs: bool = False) -> T:
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def edit_study(self, metadata: T, url: str, new: SUB_JSON) -> SUB_JSON:
+        """
+        Replace data on disk with new
+        Args:
+            metadata: study
+            url: data path to reach
+            new: new data to replace
+
+        Returns: new data replaced
+
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def check_errors(self, metadata: T) -> List[str]:
+        """
+        Check study antares data integrity
+        Args:
+            metadata: study
+
+        Returns: list of non integrity inside study
+
+        """
+        raise NotImplementedError()
```

Comment:
this is specific to raw studies. Variant studies are generated so there should be no errors there.